atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TestTransactionManagerImpl.java

```diff
@@ -103,7 +103,7 @@ public TestTransactionManagerImpl(
                 metricsManager,
                 createAssertKeyValue(keyValueService, lockService),
                 new LegacyTimelockService(timestampService, lockService, lockClient),
-                NoOpLockWatchManager.create(),
+                NoOpLockWatchManager.create(NoOpLockWatchEventCache.create()),
```

Comment:
Could you not just factor this out to be a private final field that is instantiated there and then? i.e.
`private final LockWatchEventCache eventCache = NoOpLockWatchEventCache.create();`?

Reply:
In general, you can't reference even private fields before calling a supertype constructor (which is what is happening here).

Reply:
I didn't know that. Fair enough, then - I'll approve.